From 262cb432646d611de72e9fede9aeca33e6ee7cd6 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 18 Sep 2014 14:19:36 -0400
Subject: [fs] nfsv4: set hostname when creating nfsv4 ds connection

Message-id: <1411050057-12417-14-git-send-email-steved@redhat.com>
Patchwork-id: 96181
O-Subject: [RHEL7.1 PATCH 13/94] nfsv4: set hostname when creating nfsv4 ds connection
Bugzilla: 1111708
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>

From: Peng Tao <tao.peng@primarydata.com>

We reference cl_hostname in many places for debugging purpose.
So make it useful by setting hostname when calling nfs_get_client.

Signed-off-by: Peng Tao <tao.peng@primarydata.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: a363e32e9459a0c5b199db9afa4b7dc143106012
Bugzilla: 1111708
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/nfs4client.c b/fs/nfs/nfs4client.c
index aa9ef48..53e435a 100644
--- a/fs/nfs/nfs4client.c
+++ b/fs/nfs/nfs4client.c
@@ -855,6 +855,11 @@ struct nfs_client *nfs4_set_ds_client(struct nfs_client* mds_clp,
  };
  struct rpc_timeout ds_timeout;
  struct nfs_client *clp;
+ char buf[INET6_ADDRSTRLEN + 1];
+
+ if (rpc_ntop(ds_addr, buf, sizeof(buf)) <= 0)
+  return ERR_PTR(-EINVAL);
+ cl_init.hostname = buf;
 
  /*
   * Set an authflavor equual to the MDS value. Use the MDS nfs_client
-- 
1.7.1